static/app/views/preprod/buildDetails/main/insights/appSizeInsightsSidebar.tsx

```diff
@@ -24,6 +26,19 @@ export function AppSizeInsightsSidebar({
 }: AppSizeInsightsSidebarProps) {
   const [expandedInsights, setExpandedInsights] = useState<Set<string>>(new Set());
 
+  const {
+    isHeld,
+    onDoubleClick,
+    onMouseDown,
+    size: width,
+  } = useResizableDrawer({
```

Comment:
This hook made it pretty simple to implement. The rest of the code is largely for the drag handle UI so that it plays nicely with our existing margins.